### PR TITLE
Feature/RPE-755 Change scanned date type to DateTime

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RPE-755

### Change description ###
Updated ScannedDcoument to change the scannedDate type to DateTime.
Added missing RestTemplate for Document-management-client library
Functional tests changes will be included in the next PR.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/dm/DocumentManagementConfig.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class DocumentManagementConfig {
+
+    @Bean
+    public RestTemplate restTemplate() {
+        return new RestTemplate();
+    }
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/dm/DocumentManagementConfig.java]: is this related to datetime change?

Comment (aliveni) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/dm/DocumentManagementConfig.java]: no, it is not related to DateTime change but this will fix the master build, which is failing atm
